src/java/org/apache/cassandra/schema/TableMetadata.java

```diff
@@ -461,57 +461,109 @@ public static boolean isSuperColumnMapColumnName(ByteBuffer columnName)
+    {
+        if (isIndex())
+            return;
+
+        validateTableName(previous);
+        validateTableType(previous);
+        validatePartitionKeyColumnTypes(previous);
+        validateClusteringColumns(previous);
+        validateRegularAndStaticColumns(previous);
+    }
+
     void validateCompatibility(TableMetadata previous)
     {
         if (isIndex())
             return;
 
+        validateKeyspaceName(previous);
+        validateTableName(previous);
+        validateTableId(previous);
+        validateTableType(previous);
+        validatePartitionKeyColumns(previous);
+        validateClusteringColumns(previous);
+        validateRegularAndStaticColumns(previous);
+
+    }
+
+    private void validateKeyspaceName(TableMetadata previous)
+    {
         if (!previous.keyspace.equals(keyspace))
             except("Keyspace mismatch (found %s; expected %s)", keyspace, previous.keyspace);
+    }
 
+    private void validateTableName(TableMetadata previous)
+    {
         if (!previous.name.equals(name))
             except("Table mismatch (found %s; expected %s)", name, previous.name);
+    }
 
+    private void validateTableId(TableMetadata previous)
+    {
         if (!previous.id.equals(id))
             except("Table ID mismatch (found %s; expected %s)", id, previous.id);
+    }
 
+    private void validateTableType(TableMetadata previous)
+    {
         if (!previous.flags.equals(flags) && (!Flag.isCQLTable(flags) || Flag.isCQLTable(previous.flags)))
             except("Table type mismatch (found %s; expected %s)", flags, previous.flags);
+    }
 
+    private void validatePartitionKeyColumnTypes(TableMetadata previous)
+    {
+        if (!partitionKeyType.isCompatibleWith(previous.partitionKeyType))
+        {
+            except("Partition keys of different types (found %s; expected %s)",
+                    partitionKeyType,
+                    previous.partitionKeyType);
+        }
+    }
+
+    private void validatePartitionKeyColumns(TableMetadata previous)
+    {
         if (previous.partitionKeyColumns.size() != partitionKeyColumns.size())
         {
             except("Partition keys of different length (found %s; expected %s)",
-                   partitionKeyColumns.size(),
-                   previous.partitionKeyColumns.size());
+                    partitionKeyColumns.size(),
+                    previous.partitionKeyColumns.size());
         }
 
         for (int i = 0; i < partitionKeyColumns.size(); i++)
         {
             if (!partitionKeyColumns.get(i).type.isCompatibleWith(previous.partitionKeyColumns.get(i).type))
             {
                 except("Partition key column mismatch (found %s; expected %s)",
-                       partitionKeyColumns.get(i).type,
-                       previous.partitionKeyColumns.get(i).type);
+                        partitionKeyColumns.get(i).type,
+                        previous.partitionKeyColumns.get(i).type);
```

Comment:
Looks like there's an extra space at the beginning of the lines

Reply:
Reformatted the code after importing the code style file into IntelliJ, does it look right now?